blobfuse/OAuthToken.cpp

```diff
@@ -44,60 +51,137 @@ void from_json(const json &j, OAuthToken &t) {
+                    }
+                     else
+                     {
+                            syslog(LOG_ERR, "parsing expires_on failed. Blobfuse cannot auth: OAuth token is unusable: OAuth token has an expires_on date in a non-UTC format, only UTC is supported for string format dates.\n Examples of correct UTC format dates are \"2020-04-14 16:49:11.72 +0000 UTC\" and \"2020-Apr-14 16:49:11.72 +0000 UTC\" \n Cannot use expires_in as it is missing too.\n");
+                            throw std::runtime_error("Blobfuse cannot auth: OAuth token is unusable: OAuth token has an expires_on date in a non-UTC format, only UTC is supported for string format dates.\n Examples of correct UTC format dates are \"2020-04-14 16:49:11.72 +0000 UTC\" and \"2020-Apr-14 16:49:11.72 +0000 UTC\" \n Cannot use expires_in as it is missing too.\n");
+                     }
+                }
             }
-        }
-        else
+        
+        } 
+        catch(std::exception&)
         {
-            expon_failed = true;
+            fprintf(stdout, "Blobfuse cannot auth: parsing expires_on in the OAuthToken failed. It is not an integer or string\n");
+            throw std::runtime_error("Blobfuse cannot auth: OAuth token is unusable: OAuth token did not return with an expiry time of any form.");
         }
-    } catch(std::exception&){
-        expon_failed = true;
+    }
+    else
+    {
+        throw std::runtime_error("OAuth token is unusable: OAuth token did not return with an expiry time of any form.Both expires_in and expires_on are missing in the OAuth token.");
     }
 
-    try {
-        std::string not_before;
-        not_before = j.at("not_before");
-        t.not_before = std::stoi(not_before);
-    } catch(std::exception&){} // We don't particularly care about the not_before field in blobfuse.
-
-    if(expon_failed) {
-        // We can failover and set this manually via expires_in. If expires_in failed as well, there isn't much we can do.
-        if(expin_failed) {
-            throw std::runtime_error("OAuth token is unusable: OAuth token did not return with an expiry time of any form.");
-        } else {
-            time_t current_time;
-            time(&current_time);
-
-            t.expires_on = current_time + t.expires_in;
-        }
+    if (j.contains("not_before"))
+    {
+        
+            std::string not_before;
+        try {
+            not_before = j.at("not_before");
+            t.not_before = std::stoi(not_before);
+        } 
+        catch(std::exception&){
+            syslog(LOG_INFO, "Incorrect Not before date format %s", not_before.c_str());
+        } // We don't particularly care about the not_before field in blobfuse so only send an info.
     }
 
-    // TODO: Infer expries_on if expires_in is not present, vice versa
-    //  this will __PROBABLY__ never happen. But, just on the off-chance that it does, we should be able to cope.
+}
+
+bool is_dt_number(const std::string &s)

```

Comment:
What does this method do/what is it for? Could you provide a comment above this method?

Reply:
done